zerver/lib/validator.py

```diff
@@ -494,3 +494,20 @@ def check_string_or_int(var_name: str, val: object) -> Union[str, int]:
         return val
 
     raise ValidationError(_("{var_name} is not a string or integer").format(var_name=var_name))
+
+
+def check_string_in_or_url(possible_values: List[str]) -> Validator[str]:
+    def validator(var_name: str, val: object) -> str:
+        if isinstance(val, str):
+            if val in possible_values:
+                return val
+            try:
+                return check_url(var_name, val)
+            except ValidationError:
+                raise ValidationError(
+                    _("Invalid {var_name} or {var_name} is not a URL").format(var_name=var_name)
+                )
+
+        raise ValidationError(_("{var_name} is not a string").format(var_name=var_name))
+
+    return validator
```

Comment:
This is a pretty custom function; could we instead implement it as `check_or(check_string_in(...), check_url)`, with a new generic `check_or` function?  I feel like that's going to end up as a cleaner model.

Reply:
@timabbott Following your suggestion, I've added the generic `check_or` function in validator.py, and also implement the `check_or(check_string_in(...), check_url)` for `jitsi_server_url` param validation in `realm.py`. You may check it in the latest commit.

The logic for `check_or` function is trying to validate using the first validator. If the first validation failed, we'll then proceed to the second validator. For error handling, we only return the second validator's `ValidationError` if both validations failed.

Reply:
I merged this as b5847905413cb6a895d0aad71287e1cd06dea0fe, after adding the mypy configuration to have type-checking work correctly.

Reply:
I think you can reasonably do a little follow-up project from here (in a separate PR) to replace the `check_string_or_int` type functions with `check_or(check_string, check_int)`.

Reply:
> I think you can reasonably do a little follow-up project from here (in a separate PR) to replace the `check_string_or_int` type functions with `check_or(check_string, check_int)`.

Sure!